domain/modelconfig/state/state.go

```diff
@@ -49,29 +58,25 @@ func (st *State) ModelConfigHasAttributes(
 		return rval, errors.Trace(err)
 	}
 
-	binds, vals := database.SliceToPlaceholder(attrs)
-	stmt := fmt.Sprintf(`
-SELECT key FROM model_config WHERE key IN (%s)
-`, binds)
+	attrsSlice := sqlair.S(transform.Slice(attrs, func(s string) any { return any(s) }))
+	stmt, err := sqlair.Prepare(`
+SELECT &Key.key FROM model_config WHERE key IN ($S[:])
+`, sqlair.S{}, Key{})
+	if err != nil {
+		return rval, errors.Trace(err)
+	}
 
-	return rval, db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		rows, err := tx.QueryContext(ctx, stmt, vals...)
+	return rval, db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		keys := []Key{}
+		err := tx.Query(ctx, stmt, attrsSlice).GetAll(&keys)
 		if err != nil {
-			return fmt.Errorf("deducing model config attrs set: %w", err)
+			return fmt.Errorf("getting model config attrs set: %w", err)
 		}
-		defer rows.Close()
 
-		var key string
-		for rows.Next() {
-			if err := rows.Scan(&key); err != nil {
-				return fmt.Errorf(
-					"scanning model config attribute into result set: %w",
-					err,
-				)
-			}
-			rval = append(rval, key)
+		for _, key := range keys {
```

Comment:
I've changed it too `rval = make([]string, len(keys), len(keys))` and added the keys directly. I'm hoping that should be equivalent if not faster since it avoids append.